src/components/course-settings/rename-course.cjsx

```diff
@@ -0,0 +1,94 @@
+React = require 'react'
+BS = require 'react-bootstrap'
+_ = require 'underscore'
+{CourseStore, CourseActions} = require '../../flux/course'
+{TutorInput} = require '../tutor-input'
+
+RenameCourseField = React.createClass
+
+  displayName: 'RenameCourseField'
+  propTypes:
+    courseId: React.PropTypes.string
+    label: React.PropTypes.string.isRequired
+    name: React.PropTypes.string.isRequired
+    default: React.PropTypes.string.isRequired
+    onChange: React.PropTypes.func.isRequired
+    autofocus: React.PropTypes.bool
+    validate: React.PropTypes.func.isRequired
+
+  componentDidMount: ->
+    @refs.input.focus() if @props.autofocus
+    @refs.input.cursorToEnd() if @props.autofocus
+
+  onChange: (value) ->
+    @props.onChange(value)
+
+  render: ->
+    <TutorInput
+      ref="input"
+      label={@props.label}
+      default={@props.default}
+      required={true}
+      onChange={@onChange}
+      validate={@props.validate}
+      />
+
+module.exports = React.createClass
+  displayName: 'RenameCourse'
+  propTypes:
+    courseId: React.PropTypes.string.isRequired
+
+  getInitialState: ->
+    course_name: @props.course.name
+
+  validate: (name) ->
+    error = CourseStore.validateCourseName(name, @props.course.name)
+    @setState({invalid: error?})
+    error
+
+  performUpdate: ->
+    if not @state.invalid
+      CourseActions.save(@props.courseId, course: {name: @state.course_name})
+      @refs.overlay.hide()
+
+  renderForm: ->
+    formClasses = ['modal-body', 'teacher-edit-course-form']
+    if @state?.invalid
+      formClasses.push('is-invalid-form')
+      disabled = true
+
+    <BS.Modal
+      {...@props}
+      title={'Rename Course'}
+      className='teacher-edit-course-modal'>
+
+      <div className={formClasses.join(' ')} >
```

Comment:
You should be able to use the `classnames` package from now on instead of concatenating strings